src/main/java/com/at/ac/tuwien/sepm/ss15/edulium/dao/impl/DBSectionDAO.java

```
@@ -0,0 +1,289 @@
+package com.at.ac.tuwien.sepm.ss15.edulium.dao.impl;
+
+/**
+ * Created by Administrator on 06.05.2015.
+ */
+
+import com.at.ac.tuwien.sepm.ss15.edulium.dao.DAO;
+import com.at.ac.tuwien.sepm.ss15.edulium.dao.DAOException;
+import com.at.ac.tuwien.sepm.ss15.edulium.domain.Section;
+import com.at.ac.tuwien.sepm.ss15.edulium.domain.User;
+import com.at.ac.tuwien.sepm.ss15.edulium.domain.history.History;
+import com.at.ac.tuwien.sepm.ss15.edulium.domain.validation.ValidationException;
+import com.at.ac.tuwien.sepm.ss15.edulium.domain.validation.Validator;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.core.context.SecurityContextHolder;
+
+import javax.sql.DataSource;
+import java.sql.*;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * H2 Database Implementation of the section interface
+ */
+public class DBSectionDAO implements DAO<Section> {
+    private static final Logger LOGGER = LogManager.getLogger(DBSectionDAO.class);
+
+    @Autowired
+    private DataSource dataSource;
+    @Autowired
+    private DAO<User> userDAO;
+    @Autowired
+    private Validator<Section> validator;
+
+    /**
+     * writes the object into the database and sets the identity parameter of
+     * section
+     * @param section object to store
+     * @throws DAOException if an error accessing the database occurred
+     */
+    @Override
+    public void create(Section section) throws DAOException, ValidationException {
+        LOGGER.debug("entering create with parameters " + section);
+        assert(section != null);
```

remove, validator should throw an validation exception when section == null